### PR TITLE
jailhouse-imx: upgrade to 6.1.36-2.1.0

### PR DESCRIPTION
**Pull request by MrCry0**
This closes the task in https://github.com/Freescale/meta-freescale/issues/1655

Tested building for imx8mm-lpddr4-evk, imx8mq-lpddr4-wevk, imx8ulp-lpddr4-evk.

### CODE REVIEW

**Review by angolini — APPROVED** [review]
for me it's LGTM, there is only one thing that I personally don't understand. It's "jailhouse-imx: utilise installed to native sysroot files by -dev" commit log. Maybe some rewording.

**Review by kraj — COMMENTED** [review]
what happens if you add
```
inherit python3targetconfig
```

to this recipe ?

File: recipes-extended/jailhouse/jailhouse-imx_git.bb [hunk 1]
```diff
@@ -91,6 +91,7 @@ PACKAGE_BEFORE_PN = "pyjailhouse"
 FILES:${PN} += "${nonarch_base_libdir}/firmware ${libexecdir} ${sbindir} ${JH_DATADIR}"
 # Remove libdir/* appended by setuptools3-base.bbclass for module split to work correctly
 FILES:${PN}:remove = "${libdir}/*"
+FILES:${PN}-dev += "${RECIPE_SYSROOT_NATIVE}${PYTHON_SITEPACKAGES_DIR}/*"
```
Comment (kraj) [inline recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: `inherit python3targetconfig` might help

Comment (MrCry0) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: > `inherit python3targetconfig` might help

@kraj Thanks a lot, I'll try it tomorrow morning :)

Comment (MrCry0) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: @kraj Unfortunately, it doesn't help.

```
ERROR: jailhouse-imx-git-r0 do_package: QA Issue: jailhouse-imx: Files/directories were installed but not shipped in any package:
  /home
  /home/cryo
  /home/cryo/work
  /home/cryo/work/freescale
  /home/cryo/work/freescale/build-fslc-framebuffer
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/direct_url.json
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/COPYING
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/WHEEL
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/top_level.txt
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/RECORD
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/METADATA
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/INSTALLER
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse-0.12.dist-info/REQUESTED
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/extendedenum.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/config_parser.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/pci_defs.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/cell.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__init__.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/sysfs_parser.py
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/sysfs_parser.cpython-311.pyc
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/cell.cpython-311.pyc
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/config_parser.cpython-311.pyc
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/__init__.cpython-311.pyc
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/extendedenum.cpython-311.pyc
  /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/recipe-sysroot-native/usr/lib/python3.11/site-packages/pyjailhouse/__pycache__/pci_defs.cpython-311.pyc
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
jailhouse-imx: 38 installed and not shipped files. [installed-vs-shipped]
ERROR: jailhouse-imx-git-r0 do_package: Fatal QA errors were found, failing task.
ERROR: Logfile of failure stored in: /home/cryo/work/freescale/build-fslc-framebuffer/tmp/work/imx8mq_lpddr4_wevk-fslc-linux/jailhouse-imx/git/temp/log.do_package.9768
ERROR: Task (/home/cryo/work/freescale/sources/meta-freescale/recipes-extended/jailhouse/jailhouse-imx_git.bb:do_package) failed with exit code '1'
```

**Comment by MrCry0** [comment]
> for me it's LGTM, there is only one thing that I personally don't understand. It's "jailhouse-imx: utilise installed to native sysroot files by -dev" commit log. Maybe some rewording.

@angolini I've rewritten the commit message. Is it better now?

File: recipes-extended/jailhouse/jailhouse-imx_git.bb (hunk 1 above)
Comment (MrCry0) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: @kraj @otavio We see a similar issue in our project LmP with the latest oe.
Trying to find out the root cause may take a lot of time, so I'd suggest using the hack for now and scheduling an investigation after NY.

**Comment by MrCry0** [comment]
> @angolini Maybe something shorter? I don't know if I understand properly to give a good suggestion:
> 
> jailhouse-imx: Use native-sysroot files instead by to fix -dev

We don't need the -dev package for this recipe et al. But there are files installed in native sysroot which is used after the install stage, so we cannot just remove them in do_install. So I added -dev package and packed there these files to avoid yocto building errors.

**Comment by otavio** [comment]
The install stage couldn't be fixed? Is there a pointer for it?

**Comment by MrCry0** [comment]
> The install stage couldn't be fixed? Is there a pointer for it?

I tried that way and removed these files in the install stage, but there are scripts used later so we've got another error.

If the proposed workaround is unacceptable, I could try to dedicate more time to finding another solution.

File: recipes-extended/jailhouse/jailhouse-imx_git.bb (hunk 1 above)
Comment (kraj) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: packaging native sysroot elements seems quite wrong to me. What happens if you do not package them.

Comment (otavio) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: @MrCry0, have you made any progress on this?

Comment (MrCry0) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: @otavio I haven't found another solution and will be unavailable for 5-6 weeks.
@quaresmajose said he may have time to find a more generic solution or fix it on the OE level.

Comment (quaresmajose) [reply recipes-extended/jailhouse/jailhouse-imx_git.bb:94]: https://github.com/Freescale/meta-freescale/pull/1738 should fix this issue so the line below can be removed
```
FILES:${PN}-dev += "${RECIPE_SYSROOT_NATIVE}${PYTHON_SITEPACKAGES_DIR}/*"
```

**Comment by otavio** [comment]
A rebase is needed, right?

**Comment by quaresmajose** [comment]
> A rebase is needed, right?

yup, I can take this PR and create another one

**Pull request closed without merging**